src/sdam/description/topology/server_selection/test/in_window.rs

```diff
@@ -170,19 +172,22 @@ async fn load_balancing_test() {
     ) {
         handler.clear_cached_events();
 
-        let mut handles: Vec<AsyncJoinHandle<()>> = Vec::new();
+        let mut handles: Vec<AsyncJoinHandle<Result<()>>> = Vec::new();
         for _ in 0..10 {
             let collection = client
                 .database("load_balancing_test")
                 .collection::<Document>("load_balancing_test");
             handles.push(runtime::spawn(async move {
                 for _ in 0..iterations {
-                    let _ = collection.find_one(None, None).await;
+                    collection.find_one(None, None).await?;
                 }
+                Ok(())
             }))
         }
 
-        futures::future::join_all(handles).await;
+        for handle in handles {
```

Comment:
`join_all` looks like it will await all the futures concurrently, wheras this does them in order. Since the futures being awaited here are just task handles, there isn't really a difference as they're all running in the background anyways. I think I changed this to iterate over the handles directly so I could unwrap all of them to ensure no errors occurred. This could've been done with `join_all` too, but I guess I figured I'd just do it directly instead of in two steps.